cs/safestream.texy

```diff
@@ -24,7 +24,15 @@ Může se zdát, že volání `echo 'řetězce se liší!'` nikdy nemůže nasta
 
 Uvedený kód totiž není bezpečný, pokud se zároveň provádí vícekrát (tedy ve více vláknech). Což na internetu není nic neobvyklého, často se v tentýž okamžik připojuje více lidí k jednomu webu. Takže starat se o to, aby vaše aplikace fungovala spolehlivě i při provádění ve více vláknech (thread-safe), je velmi důležité. Jak totiž vidíte, nativní PHP funkce atomické nejsou. Jinak musíme počítat se ztrátou dat a vznikem těžko odhalitelných chyb.
 
-Jak zajistit, aby se funkce `file_put_contents` nebo `file_get_contents` chovaly atomicky? Řešení nabízí bezpečný protokol SafeStream, pomocí něhož můžeme atomicky manipulovat se soubory prostřednictvím standardních PHP funkcí. Protokol se zpřístupní automaticky při načtení Nette Framework. A pak stačí jen uvést `nette.safe://` před jménem souboru:
+Jak zajistit, aby se funkce `file_put_contents` nebo `file_get_contents` chovaly atomicky? Řešení nabízí bezpečný protokol SafeStream, pomocí něhož můžeme atomicky manipulovat se soubory prostřednictvím standardních PHP funkcí. Protokol se zpřístupní automaticky při instalaci přes Composer [Composeru |composer].
```

Comment:
Udělejme to takhle, zustal bych u toho protokolu, a pak teprve šel k instalaci, tj aby to nebylo „načnutí povídání o protokolu, pak povídání o instalaci a pak teprve dokončení zmínky o protokolu“

Řešení nabízí bezpečný protokol, pomocí něhož můžeme atomicky manipulovat se soubory prostřednictvím standardních PHP funkcí. Stačí jen uvést `nette.safe://` před jménem souboru:

Příklad s nette.safe://

Protokol se zpřístupní automaticky při instalaci přes Composer [Composeru |composer]. SafeStream lze použít i bez Composeru, je ale nutné ho manualně registrovat:

Příkad na register.